### PR TITLE
Doc: make contributors and error codes valid markdown.

### PR DESCRIPTION
**Pull request by LeeWannacott**
I copied over the error codes from the website (including descriptions) which might  be overkill?, maybe you only wanted the ones already in the changelog.md :thinking: 

Also I used `<br>` to break lines instead of two white-spaces on the end because harder to see that...

GitHub Flavored Markdown Specification: https://github.github.com/gfm/

Looks like the syntax you had was for reference links:
```
[foo][bar]

[bar]: /url "title"
```
I could do it in the reference style if you want and have description as part of the error code link like it is on the website? :thought_balloon: 
It would be twice the amount of lines; although the lines for errors are really long in this PR; so it would solve that...

### CODE REVIEW

File: docs/CHANGELOG.md [hunk 1]
```diff
@@ -421,75 +421,255 @@ Beta release.
...
-[E0196]: https://quick-lint-js.com/errors/E0196/
-[E0197]: https://quick-lint-js.com/errors/E0197/
-[E0198]: https://quick-lint-js.com/errors/E0198/
-[E0199]: https://quick-lint-js.com/errors/E0199/
-[E0203]: https://quick-lint-js.com/errors/E0203/
-[E0205]: https://quick-lint-js.com/errors/E0205/
-[E0207]: https://quick-lint-js.com/errors/E0207/
+## Contributors
+
+- [AidenThing](https://github.com/AidenThing)<br>
```
Comment (strager) [inline docs/CHANGELOG.md:426]: Must fix: A visible list doesn't belong in CHANGELOG.md. We already have AUTHORS.md for the contributor list. Undo this change.

File: docs/CHANGELOG.md [hunk 2]
```diff
@@ -421,75 +421,255 @@ Beta release.
...
+- [Piotr Dąbrowski](https://github.com/yhnavein)<br>
+- [Shivam Mehta](https://github.com/maniac-en)<br>
+- [coc.nvim](https://github.com/neoclide/coc.nvim)<br>
+- [config-global-groups](https://quick-lint-js.com/config/#global-groups)<br>
+- [tiagovla](https://github.com/tiagovla)<br>
+- [wagner riffel](https://github.com/wgrr)<br>
+
+## Error codes
+
+- [E0001](https://quick-lint-js.com/errors/E0001/): variable assigned before its declaration<br>
```
Comment (strager) [inline docs/CHANGELOG.md:449]: Must fix: A list of error codes doesn't belong in CHANGELOG.md. I don't know what the purpose of this change is for. Undo.

**Review by strager — CHANGES_REQUESTED** [review]
Must fix: I don't understand the point of your change. It [break the links](https://github.com/quick-lint/quick-lint-js/blob/e6c5cb71510b24a60719882d2f85cb096b124feb/docs/CHANGELOG.md) in the content above. See screenshot:

Screen Shot 2022-03-15 at 12 32 56 PM

compared to [before your change](https://github.com/quick-lint/quick-lint-js/blob/61372e366c86e5f4c1b2cf405c060afc20da57fd/docs/CHANGELOG.md):

Screen Shot 2022-03-15 at 12 33 28 PM

> Looks like the syntax you had was for reference links:

Exactly. Why did you break this?

**Comment by LeeWannacott** [comment]
Closing PR as the changelog reference links work as intended...

**Pull request closed without merging**